pkg/collect/collector.go

```diff
@@ -286,3 +286,20 @@ func DedupCollectors(allCollectors []*troubleshootv1beta2.Collect) []*troublesho
 	}
 	return finalCollectors
 }
+
```

Comment:
Looks good to me. However, I am thinking to use pointer approach to improve
```
// EnsureCopyLast ensures Copy collectors are last in the list
// This is because copy collectors are expected to copy files from other collectors such as Exec, RunPod, RunDaemonSet.
func EnsureCopyLast(allCollectors []Collector) []Collector {
	if len(allCollectors) <= 1 {
		return allCollectors // No need to rearrange if 0 or 1 collector
	}

	// Rearrange in-place using two pointers
	lastIndex := len(allCollectors) - 1
	for i := 0; i <= lastIndex; {
		if _, ok := allCollectors[i].(*CollectCopy); ok {
			// Move the Copy collector to the end
			allCollectors[i], allCollectors[lastIndex] = allCollectors[lastIndex], allCollectors[i]
			lastIndex-- // Shrink the range for non-Copy collectors
		} else {
			i++ // Proceed with non-Copy collectors
		}
	}

	return allCollectors
}
```

Reply:
I was following other implementations for consistency for now, but definitely we can refactor it in the future if performance is a concern.

https://github.com/replicatedhq/troubleshoot/blob/67af4f0afa015451e34021d27d6acf59ae3d580c/pkg/collect/collector.go#L253-L265